### PR TITLE
Miscellaneous accumulated trivia, 202108 edition

### PR DESCRIPTION
**Pull request by nwf**
None of these are expected to have functional implications.

### CODE REVIEW

**Review by davidchisnall — APPROVED** [review]
LGTM, I'll let @mjp41 merge if he's happy with them.

File: src/test/func/memory/memory.cc [hunk 1]
```diff
@@ -255,8 +255,8 @@ void test_external_pointer()
       if ((size_t)p4 != (size_t)p1 + size - 1)
       {
         std::cout << "size: " << size << " end(p4): " << p4 << " p1: " << p1
-                  << "  p1+size-1: " << (void*)((size_t)p1 + size - 1)
-                  << std::endl;
+                  << "  p1+size-1: "
+                  << (void*)(uintptr_t)((size_t)p1 + size - 1) << std::endl;
```
Comment (mjp41) [inline src/test/func/memory/memory.cc:259]: The `(void*)` here are due to me being lazy and wanting to print the address in hex, and that is how pointers are printed.  We could fix this with `std::hex << `, or in this case actually using `pointer_offset`.
```suggestion
                  << pointer_offset(p1, size - 1) << std::endl;
```

File: src/test/func/pagemap/pagemap.cc [hunk 2]
```diff
@@ -40,9 +40,9 @@ void check_get(
 
   if (value.v != expected.v)
   {
-    std::cout << "Location: " << (void*)address << " Read: " << value.v
-              << " Expected: " << expected.v << " on " << file << ":" << lineno
-              << std::endl;
+    std::cout << "Location: " << (void*)(uintptr_t)address
```
Comment (mjp41) [inline src/test/func/pagemap/pagemap.cc:43]:
```suggestion
    std::cout << "Location: " << std::hex << address
```

**Review by mjp41 — APPROVED** [review]
LGTM couple of minor bits.

**Review by mjp41 — APPROVED** [review]

**Pull request merged**